File: jkm_production/public/js/doctype_js/journal_entry.js
// Copyright (c) 2023, Finbyz Tech PVT LTD and contributors
// For license information, please see license.txt


frappe.ui.form.on("Journal Entry", {
    onload:function(frm){
        frm.ignore_doctypes_on_cancel_all = ["Forward Contract"];
    }
});

File: jkm_production/jkm_production/doc_events/journal_entry.py
# Copyright (c) 2023, Finbyz Tech PVT LTD and contributors
# For license information, please see license.txt


import frappe
from frappe.utils import flt
from forward_contract.forward_contract.doctype.forward_contract.forward_contract import ForwardContract

def before_cancel(self, method):
    if frappe.db.exists("Forward Contract Cancellation" , {'journal_entry' : self.name}):
        fwc, parent = frappe.db.get_value("Forward Contract Cancellation", {'journal_entry' : self.name}, ["name", 'parent'])
        
        if fwc:
            frappe.db.sql(f"DELETE FROM `tabForward Contract Cancellation` WHERE name = '{fwc}'")
        
        if parent:
            doc = frappe.get_doc("Forward Contract", parent)
            ForwardContract.calculate_cancellation(doc)
            doc.save()